### DISCUSSION TITLE
There is just one token, but the iterator would split it and loop twice

### DISCUSSION THREAD

**Discussion by Amitdedhia6** [explosion/spaCy#7843] [Help: Coding & Implementations] [answered]
In the following code, I pass just one word to spacy nlp object. When we print the nlp object, it prints exactly that same word. But when we iterate the nlp object, it splits it and loops twice.

```
        nlp = spacy.load('en_core_web_lg')
        tokens = nlp('wont')
        print(tokens)               # prints wont  
        
        for token in tokens:
            print(token, token.text)    # prints two tokens -> wo and nt
```

Can anyone explain the reason behind this? Is this expected behavior (why if it is so) OR error?

I reported this as an issue as well: https://github.com/explosion/spaCy/issues/7844

**Comment by polm** [comment, marked as answer]
This is not an error. 

When you print the `tokens`, that's a Doc in spaCy, and so it just prints the original text, regardless of how it's tokenized. When you iterate over the tokens, each is printed separately.

I'm not sure what you're trying to do, but note that the expectation is that you will pass a sentence or longer input to the `nlp` function in spaCy, not an individual word.

**Comment by svlandeg** [comment]
This happens for a good reason, which is probably more clear if you also print the `token.norm_`:
```
    for token in tokens:
        print(token.text, token.norm_)
```
gives you 

> wo will
nt not

i.e. spaCy recognizes "wont" as "won't" which is conceptually two tokens - "will" and "not".

**Reply by Amitdedhia6** [reply]
Thank you - that was very helpful
